reana_client/utils.py

```diff
@@ -173,6 +180,49 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def _validate_environment_tags(reana_yaml):
+    """Validate environments in REANA specification file according to workflow type.
+
+    :param reana_yaml: Dictionary which represents REANA specifications file.
+    :return: True if environment tags are valid.
+    """
+    workflow_type = reana_yaml["workflow"]["type"]
+
+    if workflow_type == "serial":
+        workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
+        _validate_serial_workflow_environment(workflow_steps)
+    elif workflow_type == "yadage":
+        logging.info("Not Implemented yet.")
+    elif workflow_type == "cwl":
+        logging.info("Not Implemented yet.")
+
+    return True
+
+
+def _validate_serial_workflow_environment(workflow_steps):
+    """Validate environments in REANA serial workflow.
+
+    :param workflow_steps: List of dictionary which represents different steps involved in workflow.
+    :raises Warning: Warns user if valid tags are not used in serial workflow steps.
+    """
+    for step in workflow_steps:
+        environment = step["environment"].split(":")
+        environment_tag = environment[-1]
+        if len(environment) == 1:
+            logging.warning(
+                "Include a tag in the enviornment {}".format(environment[0])
+            )
+            sys.exit(1)
+        elif len(environment) == 2 and environment_tag in WARNING_TAGS:
+            logging.warning(
+                "Using {} in {} is not recommended.".format(
+                    environment_tag, environment[0]
+                )
+            )
+            sys.exit(1)
+    return True
```

Comment:
Also, when everything passed, shall we print out a message at the end that things are fine?

Otherwise currently people see:

```console
$ reana-client validate --environments                       
File .../reana-demo-root6-roofit/reana.yaml is a valid REANA specification file.
```

so they won't have any explicit output about whether image checking passed.

I guess we can emit a final message that environment images were found to be OK.  "Explicit is better than implicit", as per the Zen of Python :wink: Since we print one validation message already, we could print more for each additional image check.

P.S. This consideration may lead to a configurable "verbosity" level option, for those people that would like to go quiet, in a kind of linter style. But we could address this in a later PR...

Reply:
Added the message after check of every image tag.